File: lib/java/opentoken/src/main/java/com/truveta/opentoken/attributes/general/RecordIdAttribute.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.attributes.general;

/**
 * Represents an attribute for handling record identifiers.
 * 
 * This class extends StringAttribute and provides functionality for working with
 * identifier fields. It recognizes "RecordId" and "Id" as valid aliases for
 * this attribute type.
 * 
 * The attribute normalizes values by trimming leading and trailing whitespace
 * (inherited from StringAttribute).
 */
public class RecordIdAttribute extends StringAttribute {

    private static final String NAME = "RecordId";
    private static final String[] ALIASES = new String[] { NAME, "Id" };

    @Override
    public String getName() {
        return NAME;
    }

    @Override
    public String[] getAliases() {
        return ALIASES;
    }

}

File: lib/python/opentoken/src/main/opentoken/attributes/general/record_id_attribute.py
"""
Copyright (c) Truveta. All rights reserved.
"""

from typing import List

from opentoken.attributes.general.string_attribute import StringAttribute


class RecordIdAttribute(StringAttribute):
    """Represents a record identifier attribute.

    This class extends StringAttribute and provides functionality for working with
    record ID fields. It recognizes "RecordId" as a valid alias for this
    attribute type.

    The attribute normalizes values by trimming whitespace (inherited from StringAttribute).
    It validates that the value is not null or empty using the default BaseAttribute 
    validation rules.
    """

    NAME = "RecordId"
    ALIASES = [NAME, "Id"]

    def __init__(self):
        # Use validation rules from StringAttribute
        super().__init__()

    def get_name(self) -> str:
        """Get the name of the attribute.

        Returns:
            str: The name "RecordId"
        """
        return self.NAME

    def get_aliases(self) -> List[str]:
        """Get the aliases for the attribute.

        Returns:
            List[str]: A list containing the aliases for this attribute
        """
        return self.ALIASES.copy()
